src/PHPCensor/Controller/WidgetBuildErrorsController.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace PHPCensor\Controller;
+
+use b8;
+use PHPCensor\BuildFactory;
+use PHPCensor\Helper\Lang;
+use PHPCensor\Model\Build;
+use PHPCensor\Controller;
+
+/**
+ * Widget Build Errors Controller
+ */
+class WidgetBuildErrorsController extends Controller
+{
+    /**
+     * @var \PHPCensor\Store\BuildStore
+     */
+    protected $buildStore;
+
+    /**
+     * @var \PHPCensor\Store\ProjectStore
+     */
+    protected $projectStore;
+
+    /**
+     * Initialise the controller, set up stores and services.
+     */
+    public function init()
+    {
+        $this->buildStore = b8\Store\Factory::getStore('Build');
+        $this->projectStore = b8\Store\Factory::getStore('Project');
+    }
+
+    /**
+    * Display dashboard:
+    */
+    public function index()
+    {
+        $builds = $this->buildStore->getAllProjectsLatestBuilds();
+
+        $this->view->builds = $builds['latest'];
+
+        $view = new b8\View('WidgetBuildErrors/update');
+        $view->builds = $builds['projects'];
+        $view->projects = $this->projectStore->getByIds(array_keys($builds['projects']));
```

Comment:
We need to filter archived projects somewhere here. Like in `WidgetAllProjectsController.php` (Line 107).

Reply:
Filter added (at contoller level, not store level, to avoid join)

Reply:
Thank you, controller level is ok.